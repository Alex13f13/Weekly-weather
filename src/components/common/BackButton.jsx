import KeyboardBackspaceIcon from "@mui/icons-material/KeyboardBackspace";
import IconButton from "@mui/material/IconButton";
import { useNavigate } from "react-router-dom";

export default function BackButton() {
	const navigate = useNavigate();
	const handleGoBack = () => {
		navigate(-1);
	};

	return (
		<IconButton onClick={handleGoBack} aria-label="back" size="large">
			<KeyboardBackspaceIcon />
		</IconButton>
	);
}
