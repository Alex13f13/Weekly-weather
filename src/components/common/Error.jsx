import styled from "styled-components";

const ErrorNoDataContainer = styled.div`
	width: 90%;
	height: auto;
	background-color: #f8d7da;
	color: #721c24;
	padding: 10px;
	border: 1px solid #f5c6cb;
	border-radius: 4px;
	margin-top: 20px;
`;

const ErrorMessage = styled.p`
	margin: 0;
	text-align: center;
`;

export default function ErrorNoData() {
	return (
		<ErrorNoDataContainer>
			<ErrorMessage>Hubo un error al cargar los datos.</ErrorMessage>
		</ErrorNoDataContainer>
	);
}
