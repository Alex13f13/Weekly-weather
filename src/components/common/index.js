export { default as BackButton } from './BackButton';
export { default as DataDisplay } from './DataDisplay';
export { default as Error } from './Error';
export { default as FavButton } from './FavButton';
export { default as Loading } from './Loading';
export { default as NoResults } from './NoResults';
export { default as Selector } from './Selector';
export { default as StatusManager } from './StatusManager';
export { default as WeatherCard } from './WeatherCard';
export { default as WeatherCardList } from './WeatherCardList';
