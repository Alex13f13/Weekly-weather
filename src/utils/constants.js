export const HTML_TAGS = {
	p: "p",
	span: "span",
	div: "div",
	h1: "h1",
	h2: "h2",
	h3: "h3",
	h4: "h4",
	h5: "h5",
	h6: "h6",
};

export const SIZE = {
  small: "small",
  medium: "medium",
  large: "large",
};

export const LABEL = {
	back: "back",
	favourite: "favourite",
	max: "MAX:",
	min: "MIN:",
	province: "Provincia",
	temperature: "Temperatura:",
	temperatureMax: "Temp. M치xima:",
	temperatureMin: "Temp. M칤nima:",
	humidity: "Humedad:",
	uv: "칈ndice max. uv:",
	wind: "Viento:",
	loading: "Cargando...",
	error: "Hubo un error al cargar los datos",
	noResults: "No hay resultados",
	seletProvince: "Selecciona una provincia 游깴",
	favListButton: "Lista de favoritos",
	nextDays: "Pr칩ximos d칤as",
  seeMore: "Ver m치s",
};

export const UNITS = {
  celsius: "춿C",
  percent: "%",
  kmh: "km/h",
};

export const VARIANT = {
	primary: "primary",
	secondary: "secondary",
	success: "success",
	danger: "danger",
	warning: "warning",
	info: "info",
	standard: "standard",
  outlined: "outlined",
};

export const PROVINCE_ID = {
	Almer칤a: 1,
	Albacete: 3,
	Alicante: 4,
	츼vila: 5,
	Badajoz: 6,
	Baleares: 7,
	Barcelona: 8,
	Burgos: 9,
	C치ceres: 10,
	C치diz: 11,
	Castell칩n: 12,
	Ciudad_Real: 13,
	C칩rdoba: 14,
	A_Coru침a: 15,
	Cuenca: 16,
	Girona: 17,
	Granada: 18,
	Guadalajara: 19,
	Gipuzkoa: 20,
	Huelva: 21,
	Huesca: 22,
	Ja칠n: 23,
	Le칩n: 24,
	Lleida: 25,
	La_Rioja: 26,
	Lugo: 27,
	Madrid: 28,
	M치laga: 29,
	Murcia: 30,
	Navarra: 31,
	Ourense: 32,
	Asturias: 33,
	Palencia: 34,
	Las_Palmas: 35,
	Pontevedra: 36,
	Salamanca: 37,
	Santa_Cruz_de_Tenerife: 38,
	Cantabria: 39,
	Segovia: 40,
	Sevilla: 41,
	Soria: 42,
	Tarragona: 43,
	Teruel: 44,
	Toledo: 45,
	Valencia: 46,
	Valladolid: 47,
	Vizcaya: 48,
	Zamora: 49,
	Zaragoza: 50,
	Ceuta: 51,
	Melilla: 52,
};

