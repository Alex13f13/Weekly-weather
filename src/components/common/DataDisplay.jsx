import { createElement } from "react";
import { HTML_TAGS } from "../../utils/constants";

export default function DataDisplay({ data, label = "", units = "", tag = HTML_TAGS.p }) {
	const selectedTag = HTML_TAGS[tag] || HTML_TAGS.p;

	return data && createElement(selectedTag, null, `${label} ${data} ${units}`);
}
