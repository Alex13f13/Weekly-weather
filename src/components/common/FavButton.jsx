import IconButton from "@mui/material/IconButton";
import FavoriteIcon from "@mui/icons-material/Favorite";
import HeartBrokenIcon from "@mui/icons-material/HeartBroken";
import { SIZE, LABEL } from "../../utils/constants";

export default function FavButton({ isFavorite, onClick = () => {} }) {
	return (
		<IconButton aria-label={LABEL.favourite} size={SIZE.large} onClick={onClick}>
			{isFavorite ? <HeartBrokenIcon /> : <FavoriteIcon />}
		</IconButton>
	);
}
