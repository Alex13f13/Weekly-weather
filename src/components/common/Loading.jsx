import { LABEL } from "../../utils/constants";

export default function Loading() {
	return <>{LABEL.loading}</>;
}
