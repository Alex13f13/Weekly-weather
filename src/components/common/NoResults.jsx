import { LABEL } from "../../utils/constants";

export default function NoResults() {
	return <p>{LABEL.noResults}</p>;
}
