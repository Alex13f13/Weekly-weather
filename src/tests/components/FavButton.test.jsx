import { render, screen, fireEvent } from "@testing-library/react";
import { FavButton } from "../../components/common";
import { LABEL } from "../../utils/constants";
import { vi } from "vitest";

describe("FavButton Component", () => {
	it("renders the favorite button initially", () => {
		render(<FavButton isFavorite={false} />);

		const favoriteButton = screen.getByLabelText(LABEL.favourite);
		expect(favoriteButton).toBeDefined();
	});

	it("renders the heart broken button when marked as favorite", () => {
		render(<FavButton isFavorite={true} />);

		const heartBrokenButton = screen.getByLabelText(LABEL.favourite);
		expect(heartBrokenButton).toBeDefined();
	});

	it("calls the provided onClick function when clicked", () => {
		const onClickMock = vi.fn();

		render(<FavButton isFavorite={false} onClick={onClickMock} />);

		const favoriteButton = screen.getByLabelText(LABEL.favourite);
		fireEvent.click(favoriteButton);

		expect(onClickMock).toHaveBeenCalled();
	});

	it("matches snapshot", () => {
		const { container } = render(<FavButton isFavorite={false} />);
		expect(container).toMatchSnapshot();
	});
});
