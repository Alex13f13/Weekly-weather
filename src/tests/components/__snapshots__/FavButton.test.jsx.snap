// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FavButton Component > matches snapshot 1`] = `
<div>
  <button
    aria-label="favourite"
    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
    tabindex="0"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="FavoriteIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
      />
    </svg>
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
